packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -775,6 +783,58 @@ - (void)invalidateLayer
     _filterLayer.zPosition = CGFLOAT_MAX;
     [self.layer addSublayer:_filterLayer];
   }
+  
+  [self clearExistingGradientLayers];
+  auto background = _props->background;
+  if (background.empty()) {
+    return;
+  }
+  for (const auto &gradientPrimitive : background) {
+    CAGradientLayer *gradientLayer = [CAGradientLayer layer];
+    NSMutableArray *colors = [NSMutableArray array];
+    NSMutableArray *locations = [NSMutableArray array];
+    for (const auto& colorStop : gradientPrimitive.colorStops) {
+      if (colorStop.position.has_value()) {
+        auto location = @(colorStop.position.value());
+        UIColor* color = RCTUIColorFromSharedColor(colorStop.color);
+        [colors addObject:(id) color.CGColor];
+        [locations addObject:location];
+      }
+    }
+    gradientLayer.startPoint = CGPointMake(gradientPrimitive.startX, gradientPrimitive.startY);
+    gradientLayer.endPoint = CGPointMake(gradientPrimitive.endX, gradientPrimitive.endY);
+    
+    if (locations.count > 0) {
+      gradientLayer.locations = locations;
+    }
+    gradientLayer.colors = colors;
+    gradientLayer.frame = layer.bounds;
+    
+    // So that border layer is always above gradient layer
+    gradientLayer.zPosition = _borderLayer.zPosition;
+    [self.layer insertSublayer:gradientLayer atIndex:0];
+    
+    // Handle borders for gradient views
+    if (useCoreAnimationBorderRendering) {
+      gradientLayer.borderWidth = (CGFloat)borderMetrics.borderWidths.left;
+      CGColorRef borderColor = RCTCreateCGColorRefFromSharedColor(borderMetrics.borderColors.left);
+      gradientLayer.borderColor = borderColor;
+      CGColorRelease(borderColor);
+      gradientLayer.cornerRadius = (CGFloat)borderMetrics.borderRadii.topLeft;
+      gradientLayer.cornerCurve = CornerCurveFromBorderCurve(borderMetrics.borderCurves.topLeft);
+      gradientLayer.shouldRasterize = YES;
+      gradientLayer.drawsAsynchronously = YES;
+    } else {
+      CAShapeLayer* maskLayer = [CAShapeLayer layer];
+      CGPathRef path = RCTPathCreateWithRoundedRect(self.bounds, RCTGetCornerInsets(
+        RCTCornerRadiiFromBorderRadii(borderMetrics.borderRadii), 
+        UIEdgeInsetsZero), nil);
+      maskLayer.path = path;
+      CGPathRelease(path);
+      gradientLayer.mask = maskLayer;
+    }
+    [_gradientLayers addObject:gradientLayer];
```

Comment:
I am a bit confused why this layer has a border. Borders will go over the background, so as long as this layer is under the border layer it should be fine. Is this more about how the color is distributed when there is a border that pushes the content inward?